std/range/primitives.d

```diff
@@ -1987,6 +1993,136 @@ ElementType!R moveAt(R, I)(R r, I i) if (isIntegral!I)
 }
 
 /**
+Pops the first element from a range or array $(D r) and returns it.
+It combines $(D front) and $(D popFront) in one method.
+
+Params:
+    r = range to consume the first element from
+
+Warning:
+    If a transient range is used, you should `dup`licate it to avoid it from
+    being overwritten.
+    Using shift with `popFront` that might `throw` could lead to a loss of the
+    `front` element and thus it is prohibited.
+
+Returns:
+    Next element of the input range
+ */
+auto shift(Range)(ref Range r)
+if (isInputRange!Range
+    && functionAttributes!(popFrontN!Range) & FunctionAttribute.nothrow_)
```

Comment:
Not a typo - `popFrontN` does the attribute inference for us - just `popFront` wouldn't work due to the difference between attribute and function.